src/client/QXmppTrustStorage.h

```diff
@@ -20,43 +18,44 @@ class QXMPP_EXPORT QXmppTrustStorage
     ///
     enum SecurityPolicy {
         NoSecurityPolicy,  ///< New keys must be trusted manually.
-        Toakafa,           ///< New keys are trusted automatically until the first authentication but automatically distrusted afterwards.
+        Toakafa,           ///< New keys are trusted automatically until the first authentication but automatically distrusted afterwards. \see \xep{0450, Automatic Trust Management (ATM)}
     };
 
     ///
     /// Trust level of public long-term keys used by end-to-end encryption
     /// protocols
     ///
     enum TrustLevel {
-        AutomaticallyDistrusted = 1,  ///< The key is automatically distrusted (e.g., by ATM's security policy).
-        ManuallyDistrusted = 2,       ///< The key is manually distrusted (e.g., by clicking a button or ATM).
-        AutomaticallyTrusted = 4,     ///< The key is automatically trusted (e.g., by the client for all keys of a bare JID until one of it is authenticated).
-        ManuallyTrusted = 8,          ///< The key is manually trusted (e.g., by clicking a button).
-        Authenticated = 16,           ///< The key is authenticated (e.g., by QR code scanning or ATM).
+        Undecided = 1,                ///< The key's trust is not decided.
+        AutomaticallyDistrusted = 2,  ///< The key is automatically distrusted (e.g., by the security policy TOAKAFA). \see SecurityPolicy
+        ManuallyDistrusted = 4,       ///< The key is manually distrusted (e.g., by clicking a button or \xep{0450, Automatic Trust Management (ATM)}).
+        AutomaticallyTrusted = 8,     ///< The key is automatically trusted (e.g., by the client for all keys of a bare JID until one of it is authenticated).
+        ManuallyTrusted = 16,         ///< The key is manually trusted (e.g., by clicking a button).
+        Authenticated = 32,           ///< The key is authenticated (e.g., by QR code scanning or \xep{0450, Automatic Trust Management (ATM)}).
     };
     Q_DECLARE_FLAGS(TrustLevels, TrustLevel)
 
     virtual ~QXmppTrustStorage() = default;
 
-    virtual QFuture<void> setSecurityPolicies(const QString &encryption = {}, SecurityPolicy securityPolicy = SecurityPolicy::NoSecurityPolicy) = 0;
+    virtual QFuture<void> setSecurityPolicy(const QString &encryption, SecurityPolicy securityPolicy = SecurityPolicy::NoSecurityPolicy) = 0;
     virtual QFuture<SecurityPolicy> securityPolicy(const QString &encryption) = 0;
 
-    virtual QFuture<void> addOwnKey(const QString &encryption, const QByteArray &keyId) = 0;
-    virtual QFuture<void> removeOwnKey(const QString &encryption) = 0;
+    virtual QFuture<void> setOwnKey(const QString &encryption, const QByteArray &keyId = {}) = 0;
     virtual QFuture<QByteArray> ownKey(const QString &encryption) = 0;
```

Comment:
Currently `setOwnKey("encryption:xml-ns");` would reset the key, which looks a bit weird. How about this?:
```suggestion
    virtual QFuture<QByteArray> ownKey(const QString &encryption) = 0;
    virtual QFuture<void> setOwnKey(const QString &encryption, const QByteArray &keyId) = 0;
    inline QFuture<void> resetOwnKey(const QString &encryption) { return setOwnKey(encryption, QByteArray()); }
```